src/js/components/CoursesList/CoursesList.js

```diff
@@ -1,26 +1,46 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
+import { getCourses } from '../../services';
+import { parseSecToMinutes, parseDate } from '../../helpers';
 import styles from './CoursesList.less';
 import ListItem from '../ListItem';
 import Button from '../Button';
-import template from './template';
 
 const renderList = temp => temp.map(item => (
-  <ListItem title={item.title} timing={item.timing} text={item.text} key={item.id} />));
+  <ListItem
+    title={item.name}
+    timing={parseSecToMinutes(item.length)}
+    text={item.description}
+    key={item.id}
+    date={parseDate(item.date)}
+  />
+));
+class CoursesList extends PureComponent {
+  state = {
+    courses: []
+  }
 
-const CoursesList = () => (
-  <div className={styles.container}>
-    <span className={styles.header}>
-    Courses
-    </span>
-    {renderList(template)}
-    <Button
-      variant="contained"
-      color="primary"
-      fullWidth
-    >
-    Load more
-    </Button>
-  </div>
-);
+  componentDidMount() {
+    getCourses()
+      .then(courses => this.setState({ courses }));
+  }
+
+  render() {
+    return (
```

Comment:
what is displayed when the date is loaded?
maybe show some kind of loader or nothing a good idea, what do you think?

Reply:
Loader added)